desktop/core/src/desktop/lib/raz/raz_client.py

```diff
@@ -165,24 +167,54 @@ def check_access(self, method, url, params=None, headers=None):
           if signed_response is not None:
             return dict([(i.key, i.value) for i in signed_response.signer_generated_headers])
 
-  def _make_adls_request(self, request_data, path, resource_path):
+
+  def _make_adls_request(self, request_data, method, path, url_params, resource_path):
     storage_account = path.netloc.split('.')[0]
-    container, relative_path = resource_path.split('/', 1)
+    resource_path = resource_path.split('/', 1)
+
+    container = resource_path[0]
+    relative_path = "/"
+
+    if len(resource_path) == 2:
+      relative_path += resource_path[1]
+
+    req_params = self.handle_adls_req_mapping(method, url_params, relative_path)
 
     request_data.update({
       "clientType": "adls",
       "operation": {
         "resource": {
           "storageaccount": storage_account,
           "container": container,
-          "relativepath": relative_path,
+          "relativepath": req_params.get('relative_path'),
         },
-        "resourceOwner": storage_account,
-        "action": "read",
-        "accessTypes":["read"]
+        "action": req_params.get('access_type'),
+        "accessTypes": [req_params.get('access_type')]
       }
     })
 
+
+  def handle_adls_req_mapping(self, method, params, relative_path):
+    access_type = ''
+
+    if method == 'HEAD':
+      # Stats
+      if params.get('action') == 'getStatus':
+        access_type = 'get-status'
+    
+    if method == 'GET':
+      # Read
+      access_type = 'read'
+
+      # List
+      if params.get('resource') == 'filesystem':
+        if params.get('directory'):
+          relative_path += lib_urlunquote(params['directory'])
+          access_type = 'list'
+
+    return {'access_type': access_type, 'relative_path': relative_path}
```

Comment:
FYI: for later, usually better to fail early if those vars were not updated (and the mapping should be a var inside this function, {action --> type}